core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusService.js

```diff
@@ -135,58 +203,41 @@ oppia.factory('TranslationStatusService', [
       if (contentIdList) {
         for (var index in contentIdList) {
           contentId = contentIdList[index];
-          if (contentIdsToAudioTranslations
-            .getAudioLanguageCodes(contentId).indexOf(langCode) > -1) {
-            var audioTranslation = contentIdsToAudioTranslations
-              .getAudioTranslation(contentId, langCode);
-            if (audioTranslation.needsUpdate) {
-              return true;
-            }
+          var availabilityStatus = _getActiveStateContentAvailabilityStatus(
+            contentId);
+          if (availabilityStatus[1]) {
+            return true;
           }
         }
       }
       return false;
     };
 
     var _getActiveStateContentIdStatusColor = function(contentId) {
-      contentIdsToAudioTranslations =
-        StateContentIdsToAudioTranslationsService.displayed;
-      if (contentIdsToAudioTranslations) {
-        if (contentIdsToAudioTranslations
-          .getAudioLanguageCodes(contentId).indexOf(langCode) > -1) {
-          return ALL_AUDIO_AVAILABLE_COLOR;
-        } else {
-          return NO_AUDIO_AVAILABLE_COLOR;
-        }
+      var availabilityStatus = _getActiveStateContentAvailabilityStatus(
+        contentId);
+      if (availabilityStatus[0]) {
+        return ALL_AUDIO_AVAILABLE_COLOR;
+      } else {
+        return NO_AUDIO_AVAILABLE_COLOR;
       }
     };
 
     var _getActiveStateContentIdNeedsUpdateStatus = function(contentId) {
-      contentIdsToAudioTranslations =
-        StateContentIdsToAudioTranslationsService.displayed;
-      if (contentIdsToAudioTranslations) {
-        if (contentIdsToAudioTranslations
-          .getAudioLanguageCodes(contentId).indexOf(langCode) > -1) {
-          var audioTranslation = contentIdsToAudioTranslations
-            .getAudioTranslation(contentId, langCode);
-          if (audioTranslation.needsUpdate) {
-            return true;
-          } else {
-            return false;
-          }
-        }
-      }
+      var availabilityStatus = _getActiveStateContentAvailabilityStatus(
+        contentId);
+      return availabilityStatus[1];
     };
 
     return {
       getAllStatesNeedUpdatewarning: function() {
         return stateNeedsUpdateWarnings;
       },
-      getExplorationAudioRequiredCount: function() {
-        return explorationAudioRequiredCount;
+      getExplorationContentRequiredCount: function() {
```

Comment:
I think the above naming is ok. Or maybe content -> translation. Content seems misleading.

Reply:
Hmmm.. Actually, I think let's leave it as is. Adding translation in the name would cause further confusion as a voiceover isn't a translation. Though I am not super happy with the current naming, I can't think of anything better 😅.

Reply:
Can we do something like this:
```
getExplorationVoiceoverRequiredCount
getExplorationTranslationRequiredCount
```

and call whichever required?

Reply:
That would be good. Are you interested in the individual counts, or total counts? If the former I think making separate functions would make sense.

Reply:
@nithusha21, I went through this and realized we use to count this data based on the active mode as creating two functions will not be possible (As they will return the same value). Can we rename this to `getActiveModeAssetsRequiredCount`?

Reply:
Hmmm, then this function should take the mode it is in as a parameter? Not sure, whatever you feel is fit. I am ok with the name as is though!